### PR TITLE
PP-8435 Support new Stripe contract pricing

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID
- Changes to calculate and apply new connect fee percentage for Stripe transactions from the configured date.

Infra changes to for new env variables https://github.com/alphagov/pay-infra/pull/3214

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java [hunk 1]
```diff
@@ -56,14 +57,14 @@ public CaptureResponse capture(CaptureGatewayRequest request) {
             } else {
                 capturedCharge = captureWithChargeAPI(request);
             }
-            
+
             Optional<Long> processingFee = capturedCharge.getFee()
-                    .flatMap(fee -> calculateProcessingFee(request.getAmount(), fee));
-                    
+                    .flatMap(fee -> calculateProcessingFee(request.getCreateDate(), request.getAmount(), fee, stripeGatewayConfig));
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java]: Does `stripeGatewayConfig` need to be passed in as a parameter? Can't we just use it from the field on the class?

File: src/main/java/uk/gov/pay/connector/gateway/model/request/CaptureGatewayRequest.java [hunk 2]
```diff
@@ -31,6 +32,10 @@ public String getExternalId() {
         return charge.getExternalId();
     }
 
+    public Instant getCreateDate() {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/model/request/CaptureGatewayRequest.java]: should this be `getCreatedDate`?

**Review by stephencdaly — COMMENTED** [review]
Looks good, minor comments

File: src/main/java/uk/gov/pay/connector/gateway/model/request/CaptureGatewayRequest.java (hunk 2 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/gateway/model/request/CaptureGatewayRequest.java]: Good spot! updated

File: src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java]: Updated.

File: src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java [hunk 3]
```diff
@@ -143,9 +144,17 @@ private void transferToConnectAccount(CaptureGatewayRequest request, Long netTra
         );
     }
 
-    private Optional<Long> calculateProcessingFee(Long grossChargeAmount, Long stripeFee) {
+    private Optional<? extends Long> calculateProcessingFee(Instant createDate, Long grossChargeAmount, Long stripeFee) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java]: This parameter should probably be `createdDate` too

**Review by stephencdaly — DISMISSED** [review]

File: src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java (hunk 3 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeCaptureHandler.java]: 🤦🏽‍♂️ updated.

**Review by stephencdaly — APPROVED** [review]

**Pull request merged**